src/sessionTypes/passwordGrantAuth.js

```diff
@@ -0,0 +1,182 @@
+   * @fulfills {string} apiToken
+   * @rejects {Error}
+   */
+  getCurrentApiToken() {
+    return new Promise((resolve, reject) => {
+      if (!this._sessionInfo.accessToken) {
+        return reject(new Error('No access token found.'));
+      }
+
+      return resolve(this._sessionInfo.accessToken);
+    });
+  }
+
+  /**
+   * Tells caller if the current user is authenticated
+   *
+   * @returns {boolean}
+   */
+  isAuthenticated() {
+    const hasToken = !!(this._sessionInfo && this._sessionInfo.accessToken);
+
+    return hasToken;
+  }
+
+  /**
+   * Logs the user in using Auth0 using a username a password
+   *
+   * @param {string} username The username of the user to authenticate
+   * @param {string} password The password of the user to authenticate
+   *
+   * @returns {Promise}
+   * @fulfills {string}
+   * @rejects {Error}
+   */
+  logIn(username, password) {
+    return new Promise((resolve, reject) => {
+      const audience = this._sdk.config.audiences.contxtAuth.clientId;
+
+      this._auth0.loginWithDefaultDirectory(
+        { password, username, audience },
+        (err, response) => {
+          if (err) {
+            const errorMessage =
+              (err && err.description) || 'Authentication failed.';
+
+            return reject(new Error(errorMessage));
+          }
+
+          return resolve(response.accessToken);
+        }
+      );
+    }).then((accessToken) => {
+      return this._getApiToken(accessToken);
+    });
+  }
+
+  /**
+   * Logs the user out by removing any stored session info.
+   *
+   * @returns {Promise}
+   * @fulfills {string}
+   */
+  logOut() {
+    return new Promise((resolve) => {
+      this._sessionInfo = {};
+
+      return resolve('Logout successful - session info cleared.');
+    });
+  }
+
+  /**
+   * Requests an access token from Contxt Auth with the correct audiences.
+   *
+   * @param {string} accessToken
+   *
+   * @returns {Promise}
+   * @fulfill {string} accessToken
+   * @rejects {Error}
+   *
+   * @private
+   */
+  _getApiToken(accessToken) {
+    return axios
+      .post(
+        `${this._sdk.config.audiences.contxtAuth.host}/v1/token`,
+        {
+          audiences: Object.keys(this._sdk.config.audiences)
+            .map((audienceName) => {
+              return this._sdk.config.audiences[audienceName].clientId;
+            })
+            .filter((clientId) => {
+              return (
+                clientId &&
+                clientId !== this._sdk.config.audiences.contxtAuth.clientId
+              );
+            }),
+          nonce: 'nonce'
+        },
+        {
+          headers: { Authorization: `Bearer ${accessToken}` }
+        }
+      )
+      .then((response) => {
+        const { data } = response;
+
+        this._saveSession({
+          accessToken: data.access_token
+        });
+
+        return data.access_token;
+      })
+      .catch(() => {
+        throw new Error('An error occurred during authorization with Contxt.');
```

Comment:
What do you think about getting rid of this catch and letting the error just propagate? The WebAuth one does that and it's nice. The Machine Auth one does not, but uses a different error message: `'There was a problem getting a token from the ContxtAuth server. Please check your configuration settings.'`. If not getting rid of the catch, could you change the text?

Reply:
I like the idea of just letting it propagate. I'll drop the `catch` here similar to the WebAuth.